docs/serving/using-auto-tls.md

```diff
@@ -0,0 +1,150 @@
+---
+title: "Using Auto-TLS in Knative"
+weight: 15
+type: "docs"
+---
+
+Within Knative, we provide a feature of automatically provisioning and 
+configuring TLS certificates to terminate the external TLS connection. This doc
+provides a guide about how to enable and configure Auto TLS feature in Knative.
+
+## Prerequesties
+
+<!-- TODO(zhiminx) add the link about installing Istio with SDS enabled after PR https://github.com/knative/docs/pull/1272 is checked in-->
+1. Follow the [instructions]() to install Istio with SDS enabled.
+2. Follow the [instructions](../installing-cert-manager.md) to install Cert-Manager.
+
+## Configure Cert-Manager
+
+> Note that currently the Auto-TLS feature only supports DNS challenge.
+
+### Set up DNS challenge Provider
+
+Follow the [instructions](https://docs.cert-manager.io/en/latest/tasks/acme/configuring-dns01/index.html#supported-dns01-providers) to set up DNS challenge issuer according to the 
+DNS provider of your custom domain.
+
+#### Google Cloud DNS Provider Instruction
+
+Specifically for Google Cloud DNS, below are steps to set up the DNS challenge 
+issuer.
+
+##### Creating a Cloud DNS service account
+
+To add the TXT record, configure Knative with a service account that can be used
+by cert-manager to create and update the DNS record.
+
+To begin, create a new service account with the project role `dns.admin`:
+
+```shell
+# Set this to your GCP project ID
+export PROJECT_ID=<your-project-id>
+
+# Name of the service account you want to create.
+export CLOUD_DNS_SA=cert-manager-cloud-dns-admin
+gcloud --project $PROJECT_ID iam service-accounts \
+  create $CLOUD_DNS_SA \
+  --display-name "Service Account to support ACME DNS-01 challenge."
+
+# Fully-qualified service account name also has project-id information.
+export CLOUD_DNS_SA=$CLOUD_DNS_SA@$PROJECT_ID.iam.gserviceaccount.com
+
+# Bind the role dns.admin to this service account, so it can be used to support
+# the ACME DNS01 challenge.
+gcloud projects add-iam-policy-binding $PROJECT_ID \
+  --member serviceAccount:$CLOUD_DNS_SA \
+  --role roles/dns.admin
+
+# Download the secret key file for your service account.
+gcloud iam service-accounts keys create ~/key.json \
+  --iam-account=$CLOUD_DNS_SA
+```
+
+After obtaining the service account secret, publish it to your cluster. This
+command uses the secret name `cloud-dns-key`, but you can choose a different
+name.
+
+```shell
+# Upload that as a secret in your Kubernetes cluster.
+kubectl create secret --namespace cert-manager generic cloud-dns-key \
+  --from-file=key.json=$HOME/key.json
```

Comment:
nit: you use `$HOME` and `~` inconsistently.  

Also, consider using a less generic name for things in the home directory (or mkdtemp